modules/instrument_manager/php/NDB_Menu_Filter_instrument_manager.class.inc

```diff
@@ -96,16 +96,31 @@ class NDB_Menu_Filter_Instrument_Manager extends NDB_Menu_Filter
                     null
                 );
 
-                exec(
-                    "mysql".
-                    " -h" . escapeshellarg($db_config['host']).
-                    " -u" . escapeshellarg($db_config['username']).
-                    " -p" . escapeshellarg($db_config['password']).
-                    " " . escapeshellarg($db_config['database']).
-                    " < " . $this->path . "project/tables_sql/".
-                    $instrument->table.".sql"
-                );
-
+		if(isset($db_config['quatUser']) && isset($db_config['quatPassword'])){
+			if(!is_null($db_config['quatUser']) && !is_null($db_config['quatPassword'])){ 
```

Comment:
From [php `isset` documentation](http://php.net/manual/en/function.isset.php): "isset — Determine if a variable is set and is not NULL"

You should be able to get rid of line 100 and it will still execute the same way.

In fact, as @driusan well likely tell you, it's a bit of a nicer style to write something like

```php
if (!isset($var)...) {
    throw new Exception() {
    }
}
// SQL code goes here
```

That way you only have to write the `Exception` code once and it's a bit easier to read.

Would you mind changing the code that way?